### ISSUE TITLE
Question about free enegry uncertainties

### ISSUE THREAD

**Issue by harlor** [choderalab/pymbar#304]
I noticed that some people (for example: https://github.com/MobleyLab/alchemical-analysis/blob/37b21ebb6d0495990c43595d9ade10c779c3ab08/alchemical_analysis/alchemical_analysis.py#L650) just use the sum of the squares of uncertainties between intermediate states to calculate the uncertainty of the total difference in free energy. This requires the free energy differences to be uncorrelated correct? 

But is this really the case? We know that we (can) have nonzero contributions in the covariance matrix (eq 12 https://aip.scitation.org/doi/10.1063/1.2978177).

In my random example this makes a huge difference in the uncertainty of the free energy difference:

using (eq 12) - `np.sqrt(Theta[0, 0] - 2*Theta[0, -1] + Theta[-1, -1])`
err: 0.072118

using sum of the squares - alchemical-analysis:
err: 0.032283

**Comment by harlor** [comment]
Btw. I can repoduce the alchemical-analysis value using:
```python
var = 0.0
for i in range(K-1):
  var += Theta[i, i] - 2*Theta[i, i+1] + Theta[i+1, i+1]
print('err2: %f' % (np.sqrt(var) / B))
```
err2: 0.032283

**Comment by jchodera** [comment]
You are correct that using the sum of squares of uncertainties between sequential pairs of intermediate states requires the data used to estimate those differences to be uncorrelated! Since practitioners typically use the data for an intermediate state `i` to estimate both `(i-1, i)` and `(i,i+1)` free energies, this condition is generally violated, and the uncertainty can be underestimated.

Pymbar should give a superior estimate of the uncertainty in this case because, as you point out, it can include the covariance not only in sequential pairs of free energy differences, but in the free energy estimate obtained from all the data jointly.

**Comment by harlor** [comment]
Thanks for your fast reply! You convinced me that it is a good idea to change this.
